inject/src/main/java/io/micronaut/context/env/PropertySource.java

```diff
@@ -160,4 +245,30 @@ enum PropertyConvention {
          */
         JAVA_PROPERTIES
     }
+
+    /**
+     * The origin of the property source.
+     * @since 4.8.0
+     */
+    @Experimental
+    record Origin(@NonNull String location) {}
+
+    /**
+     * A property entry models
+     * @param property The normalized property key
+     * @param value The property value
+     * @param raw The raw property key that originated the property
+     * @param origin
+     */
+    @Experimental
+    record PropertyEntry(
```

Comment:
Can you convert it to an interface? I don't like the idea of exposing a record as a public API